apps/api/src/app/topics/e2e/rename-topic.e2e.ts

```diff
@@ -47,48 +31,61 @@ describe('Rename a topic - /topics/:topicKey (PATCH)', async () => {
   });
 
   it('should rename the topic and return it if exists in the database', async () => {
-    const url = `${BASE_PATH}/${topicKey}`;
-    const patchResponse = await session.testAgent.patch(url).send({ name: renamedTopicName });
-
-    expect(patchResponse.statusCode).to.eql(200);
-
-    const topic = patchResponse.body.data;
+    const newVar = await createTopicSubscriberAndAttach();
+    const patchResponse = await novuClient.topics.rename({ name: renamedTopicName }, topicKey);
+    const topic = patchResponse.result;
 
-    expect(topic._environmentId).to.eql(session.environment._id);
-    expect(topic._organizationId).to.eql(session.organization._id);
+    expect(topic.environmentId).to.eql(session.environment._id);
+    expect(topic.organizationId).to.eql(session.organization._id);
     expect(topic.key).to.eql(topicKey);
     expect(topic.name).to.eql(renamedTopicName);
-    expect(topic.subscribers).to.have.members([firstSubscriber.subscriberId, secondSubscriber.subscriberId]);
+    expect(topic.subscribers).to.have.members([
+      newVar.firstSubscriber.subscriberId,
+      newVar.secondSubscriber.subscriberId,
+    ]);
   });
 
   it('should throw a not found error when the topic id provided does not exist in the database', async () => {
+    await createTopicSubscriberAndAttach();
     const nonExistingId = 'ab12345678901234567890ab';
-    const { body } = await session.testAgent.patch(`${BASE_PATH}/${nonExistingId}`).send({ name: renamedTopicName });
-
-    expect(body.statusCode).to.equal(404);
-    expect(body.message).to.eql(
-      `Topic not found for id ${nonExistingId} in the environment ${session.environment._id}`
+    const { error, parsedBody } = await expectSdkExceptionGeneric(() =>
+      novuClient.topics.rename({ name: renamedTopicName }, nonExistingId)
     );
-    expect(body.error).to.eql('Not Found');
+    expect(error).to.be.ok;
+    if (error && parsedBody) {
```

Comment:
Why are these expectations conditional? Can we avoid that?

Reply:
pure for typeguard in order to avoid ? ! annotations, the expectation is already there in line 54

Reply:
it's a special exectation for an error, it handles it generically, that's the expectation I chose, to make sure there is an error avoiding 100 try catch